### DISCUSSION TITLE
Time Range Selection and Data Mismatch in Services Apdex  Success Rate  and Service Load Matrix

### DISCUSSION THREAD

**Discussion by vishalkadam18** [apache/skywalking#12441] [Q&A]
Hello Team,

When selecting a time range from 10:00 to 11:30 in the services' Service Apdex and Success Rate, Service Load Matrix, the data fetched covers the range from 10:00 to 11:59. However, when running the REST API to fetch the average data from the Elasticsearch database for the same time range (10:00 to 11:59), there is a value mismatch.

Thanks.

**Comment by wu-sheng** [comment]
Could you be more clear? Provide which query are you applying?

**Reply by vishalkadam18** [reply]
we are running this query on devtools matching the value in ui

GET /sw_metrics-*/_search?size=0
{
  "query": {
    "bool": {
      "must": [
        {
          "term": {
            "metric_table": "service_apdex"
          }
        },
        {
          "term": {
            "entity_id": "SXNoaW5lX1Byb2Q=.1"
          }
        },
        {
          "range": {
            "time_bucket": {
              "gte": "202407151000",
              "lte": "202407151159"
            }
          }
        }
      ]
    }
  },
  "aggs": {
    "avg_int_value": {
      "avg": {
        "field": "int_value"
      }
    }
  }
}

we are the dividing result by the avg(service_apdex)/10000

As my understanding there is mismatch value

**Reply by wu-sheng** [reply]
If you run elasticsearch query, that is not our concern. We are not support elasticsearch knowledge.

If you have questions about the data structure, I may be able to help.

**Reply by vishalkadam18** [reply]
Hello Team,

I am requesting clarification on the logic used to calculate and aggregate the service load, as I have observed inconsistencies when selecting different time ranges. like , when selecting a time range of 16:00 to 17:00, the service load shows 14 calls/min. However, when selecting a time range of 16:00 to 17:59, the service load shows 0 calls/min. This discrepancy makes it unclear how the service load is being calculated and aggregated over different time ranges. 

Could you provide details on the aggregation methods or algorithms used and explain the reason for these observed discrepancies? 

Your assistance is appreciated.

Thanks

**Reply by wu-sheng** [reply]
Because the metrics are calculated and queried in different dimensionalities. Such as when you have 16-17, UI is actually querying data in minute dimensionality, but when you take 16-18, then it's querying data in hour dimensionality.
Then about `cpm`, minute dimensionality mean at least one call per minute, but in the hour dimensionality, 1cpm minute there is at least one query in every minute of this hour, so, it equals to say you at least should have 60 calls per hour to reach 1 cpm in HOUR dimensionality.

This is just a statistics thing, very common. 14 calls happen in one minute of one hour, but nothing more, then, 14 cpm in MINUTE dimensionality ~= 0 cpm in HOUR dimensionality
